test/e2e/util.go

```diff
@@ -182,3 +186,121 @@ func randomSuffix() string {
 func expectNoError(err error, explain ...interface{}) {
 	ExpectWithOffset(1, err).NotTo(HaveOccurred(), explain...)
 }
+
+func cleanup(filePath string, selectors ...string) {
+	By("using stop to clean up resources")
+	runKubectl("stop", "-f", filePath)
+
+	for _, selector := range selectors {
+		resources := runKubectl("get", "pods,rc,se", "-l", selector, "--no-headers")
+		if resources != "" {
+			Failf("Resources left running after stop:\n%s", resources)
+		}
+	}
+}
+
+type validatorFn func(c *client.Client, podId string) (interface{}, error)
```

Comment:
so --- the theoretical input to a validator is (1) a client, whose state we are validating and (2) a pod Id.  in some cases (2) is not required, but i can't imagine a case where (1) is never required ?  I'm trying to stick to the "pure" implementation where the validator does *nothing* but validate the cluster state, i.e., it doesnt create a client or do any other work. does that make sense.